openmole/plugins/org.openmole.plugin.tool.pattern/src/main/scala/org/openmole/plugin/tool/pattern/Strain.scala

```diff
@@ -28,7 +29,14 @@ object Strain {
     val first = Capsule(EmptyTask(), strainer = true)
     val last = Slot(Capsule(EmptyTask(), strainer = true))
 
-    (first -- puzzle -- last) + (first -- last)
+    val mole = puzzle.toMole
+    val sources = Sources(puzzle.sources.groupBy(_._1).mapValues(_.map(_._2)))
+    val hooks = Hooks(puzzle.hooks.groupBy(_._1).mapValues(_.map(_._2)))
```

Comment:
Yeah you're right. Would you mind implementing an implicit conversion in the workflow.mole package so we don't have to write this code again and again?

The best place would be to write the implicit conversion in the Sources and Hooks companion objects (uncreated yet)?